backend/assignments/2/src/main/java/com/caching/exceptions/GlobalExceptionHandling.java

```diff
@@ -0,0 +1,31 @@
+package com.caching.exceptions;
+
+import com.caching.dto.ErrorDTO;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandling {
+    /**
+     * For any generic exception
+     * @param ex
+     * @return Response thrown by exception
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> handleGeneralException(Exception ex){
```

Comment:
More exceptions could have been exposed. Instead of wrapping almost all of them under `INTERNAL_SERVER_ERROR`. These must be limited for edge cases/missed out stuff.